apps/sms/js/thread_ui.js

```diff
@@ -2074,8 +2059,8 @@ var ThreadUI = {
           params.items.push({
             l10nId: 'forward',
             method: () => {
-              this.navigateToComposer({
-                forward: { messageId: messageId }
+              MessageManager.getMessage(messageId).then((message) => {
```

Comment:
not sure it will be simpler :) because it would likely still be asynchornous (not sure yet)